File: Data Structures/Queue.java
public class Queue{
    private static class Node{
        private int data;
        private Node next;
        private Node(int data){
            this.data = data;
        }
    }
    private Node Head;
    private Node Tail;

    // empty
    private boolean isEmpty(){
        return Head == null;
    }
    // peek function
    public int peek(){
        if(!isEmpty()){
            return Head.data;}
    }
    // add function
    public void add(int data){
        Node node = new Node(data);
        if (Head == null){
            Head = node;
            Tail = node;
        }
        else{
        Tail.next = node;
        Tail = node;
        }
    }
    // remove function 
    public int remove(){
        int data = Head.data;
        Head = Head.next;
        if(isEmpty()){
            Tail = null;
        }
        return data;
    }

}

File: Data Structures/Queue.py
# i'll make it not aa static class so there can be more than 1 Queue



class Queue:
    noOfQueues = 0
    def __init__(self):
        self.Head = None
        self.Tail = None
        Queue.noOfQueues += 1
    class Node:
        def __init__(self, data):   
            self.data = data
            self.nextNode = None



    # check empty
    def isEmpty(self):
        return self.Head == None
    # peek
    def peek(self):
        return self.Head.data
    # add
    def add(self, data):
        node = self.Node(data)
        if self.Head == None:
            self.Head = node
        if self.Tail != None:
            self.Tail.nextNode = node
        self.Tail = node
    # remove
    def remove(self):
        data = self.Head.data
        self.Head = self.Head.nextNode
        if self.Head == None:
            self.Tail == None
        return data
        
def main():
    q = Queue()
    b = Queue()
    c = Queue() 
    print(Queue.noOfQueues)
    q.add(12)
    q.add(24)
    q.add(36)
    q.add(42)
    print(q.peek())
    print(q.remove())
    print(q.peek())


if __name__ == "__main__":
    main()


